certmanager/generate_cert_test.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certmanager
+
+import (
+	"encoding/pem"
+	"crypto/rsa"
+	"crypto/x509"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+)
+
+// VerifyFields contains the certficate fields to verify in the test.
+type VerifyFields struct {
+	notBefore time.Time
+	notAfter  time.Time
+	extKeyUsage x509.ExtKeyUsage
+	isCA		bool
+	org		string
+}
+
+func TestGenCert(t *testing.T) {
+	now := time.Now()
+	nowInString := now.Format("Jan 2 15:04:05 2006")
+	// Options to generate a CA cert.
+	caCertOptions := CertOptions{
+		Host: "test_ca.com",
+		ValidFrom: nowInString,
+		ValidFor: time.Hour * 24 * 30,
+		SignerCert: nil,
+		SignerPriv: nil,
+		Org: "MyOrg",
+		IsCA: true,
+		IsSelfSigned: true,
+		IsClient: false,
+	}
+	notBefore, _ := time.Parse("Jan 2 15:04:05 2006", nowInString)
```

Comment:
Consider making the layout string a constant so you can just use it here and in the `GenCert()` method.